src/Resources/contao/models/FilesModel.php

```diff
@@ -175,6 +175,14 @@ public static function findByUuid($strUuid, array $arrOptions=array())
 			$strUuid = \StringUtil::uuidToBin($strUuid);
 		}
 
+		// Check in model registry (does not work by default due to UNHEX())
+		$objModel = \Model\Registry::getInstance()->fetch(static::$strTable, $strUuid, 'uuid');
```

Comment:
Similar to https://github.com/contao/core-bundle/blob/b894c22a85d68d2dea8fdfe3dbb7f5be1c2df38b/src/Resources/contao/models/PageModel.php#L660-L683 we should add a `empty($arrOptions)` check here, shouldn't we?